mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -255,6 +255,20 @@ private[spark] object BLAS extends Serializable with Logging {
     }    
   }
 
+  // Workaround for SparseVectors.
+  def syr(alpha: Double, x: SparseVector, A: DenseMatrix) {
+    val mA = A.numRows
+    val nA = A.numCols
+    require(mA == nA, s"A is not a square matrix. A: $mA x $nA")
```

Comment:
Hmm.  Symmetry is a requirement of the syr() function; what is being performed is an incomplete check.  The old message mismatched the check, but matched the requirement; perhaps this could be something like "A is not square, and thus cannot be symmetric" ... or something along that line?